core/src/main/scala/com/softwaremill/diffx/DiffFor.scala

```diff
@@ -1,14 +1,29 @@
 package com.softwaremill.diffx
 
-trait DiffFor[T] {
-  def apply(left: T, right: T): DiffResult
+trait DiffFor[T] { outer =>
+  protected val ignored: List[List[String]] = List.empty
+
+  def apply(left: T, right: T): DiffResult = apply(left, right, Nil)
+  def apply(left: T, right: T, toIgnore: List[List[String]]): DiffResult
+
+  private[diffx] def ignoreUnsafe(fields: String*): DiffFor[T] = new DiffFor[T] {
+    override def apply(left: T, right: T, toIgnore: List[List[String]]): DiffResult =
+      outer.apply(left, right, toIgnore ++ ignored)
+    override val ignored: List[List[String]] = outer.ignored ++ List(fields.toList)
+  }
 }
 
 object DiffFor {
   def apply[T, R: DiffFor](converter: T => R): DiffFor[T] =
```

Comment:
isn't that `map`?

Reply:
IMO it's more like `contramap` ;) Also it depends how you perceive it. I see it as a factory method which indeed under the hood does mapping. But for the user it might be more convenient to call `DiffFor[Person,Int](_.age)` rather than `implicitly[DiffFor[Int]].contramap(_.age)`

Reply:
what would `DiffFor[Person,Int](_.age)` do?

Reply:
What did you expect `map` would do?

Reply:
`val pd: Diff[Person] = implicitly[Diff[Int]].contramap((_: Person).age)` creates a diff for `Person` basing on diff for `Int`. Just as many other codecs/encoders/metas etc.

`val pd: Diff[Person] = Diff[Person, Int](_.age)` well ... I wouldn't be so sure